js/lib/beautify.js

```diff
@@ -423,7 +448,11 @@
                 return
             }
 
-            if ((opt.preserve_newlines && current_token.wanted_newline) || force_linewrap) {
+            var shouldPreserveOperatorNewline = (opt.operator_position === OPERATOR_POSITION.preserve_newline) &&
+                in_array(flags.last_text, OP_POSITION_OPERATORS) ||
+                in_array(current_token.text, OP_POSITION_OPERATORS);
+                
+            if (((opt.preserve_newlines || shouldPreserveOperatorNewline) && current_token.wanted_newline) || force_linewrap) {
```

Comment:
(overide_general_preserve) 
This seems wrong.  The default is not `remove_newline`, so if someone sets `operator_position  = remove_newline`, it should take precedence over the general `preserve_newlines = true` setting.   That is what you have it doing for `operator_position  = preserve_newline` with ``preserve_newlines = false`, so the inverse should also hold.